src/views/tutorial-view/components/next-previous/helpers.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  Collection as ClientCollection,
+  CollectionLite as ClientCollectionLite,
+  TutorialLite as ClientTutorialLite,
+} from 'lib/learn-client/types'
+import {
+  getCollectionSlug,
+  getTutorialSlug,
+} from 'views/collection-view/helpers'
+import { NextPreviousProps } from '.'
+
+interface GetNextPreviousParams {
+  currentTutorialSlug: string
+  currentCollection: ClientCollection
+  nextCollectionInSidebar: ClientCollectionLite
+}
+
+export function getNextPrevious({
+  currentTutorialSlug,
+  currentCollection,
+  nextCollectionInSidebar,
+}: GetNextPreviousParams): NextPreviousProps {
+  let previousTutorial
+  let nextTutorial
+  let nextCollection
+  const tutorialIndex = currentCollection.tutorials.findIndex(
+    (t: ClientTutorialLite) => t.slug === currentTutorialSlug
+  )
+  const isFirstTutorial = tutorialIndex === 0
+  const isLastTutorial =
+    tutorialIndex === currentCollection.tutorials.length - 1
+
+  if (!isFirstTutorial) {
+    const { slug, name } = currentCollection.tutorials[tutorialIndex - 1]
+    previousTutorial = {
+      path: getTutorialSlug(slug, currentCollection.slug),
+      name,
+    }
+  }
+
+  if (!isLastTutorial) {
+    const { slug, name } = currentCollection.tutorials[tutorialIndex + 1]
+    nextTutorial = {
+      path: getTutorialSlug(slug, currentCollection.slug),
+      name,
+    }
+  }
+
+  if (nextCollectionInSidebar) {
+    nextCollection = {
+      path: getCollectionSlug(nextCollectionInSidebar.slug),
+      name: nextCollectionInSidebar.shortName,
+    }
+  }
+
+  /**
+   * @TODO - interim state for 'final' link
```

Comment:
I dig the clarity in this comment, I was wondering about the "last collection" UX - & this seems like a great interim fix 💯